PricingLoader.scala

```diff
@@ -0,0 +1,54 @@
+package org.llm4s
+
+import scala.io.Source
+import scala.util.{Try, Using}
+import play.api.libs.json._
+
+object PricingLoader {
+  private val pricingUrl = "https://raw.githubusercontent.com/BerriAI/litellm/main/model_prices_and_context_window.json"
+  private var cachedPricing: Option[JsValue] = None
+
+  // Load pricing data from URL
+  def loadPricing(): Option[JsValue] = {
+  if (cachedPricing.isEmpty) {
+    println("Fetching pricing data from URL...")

```

Comment:
We should make each choice the logic here makes clear and informative:
e.g.
'Fetching model pricing and context window information from $url'

'Failed to load model pricing information from $url, falling back to local file.